google/cloud/aiplatform/prediction/local_model.py

```diff
@@ -0,0 +1,140 @@
+from google.cloud.aiplatform.prediction.predictor import Predictor
+from google.cloud.aiplatform.utils import prediction_utils
+
+DEFAULT_PREDICT_ROUTE = "/predict"
+DEFAULT_HEALTH_ROUTE = "/health"
+DEFAULT_HTTP_PORT = 8080
+
+
+class LocalModel:
+    """Class that represents a local model."""
+
+    def __init__(
+        self, serving_container_spec: aiplatform.gapic.ModelContainerSpec,
+    ):
+        """Creates a local model instance.
+
+        Args:
+            serving_container_spec (aiplatform.gapic.ModelContainerSpec):
+                Required. The container spec of the LocalModel instance.
+        """
+        self.serving_container_spec = serving_container_spec
+
+    @classmethod
+    def create_cpr_model(
+        cls,
+        src_dir: str,
+        output_image_uri: str,
+        predictor: Optional[Type[Predictor]] = None,
+        handler: Type[Handler] = PredictionHandler,
+        base_image: str = "python:3.7",
+        requirements_path: Optional[str] = None,
+    ) -> "LocalModel":
+        """Creates a local model from a custom predictor.
+
+        This method creates a docker image to include user-provided predictor, and handler.
+        It populates the entrypoint, entrypoint.py, right under the specified directory, src_dir,
+        if it doesn't exist and generates a Dockerfile to build the image.
+
+        An example src_dir (e.g. "./user_src_dir") provided looks like:
+        .
+        |-- user_src_dir/
+            |-- predictor.py
+            |-- requirements.txt
+            |-- user_code/
+            |   |-- utils.py
+            |   |-- custom_package.tar.gz
+            |   |-- ...
+            |-- ...
+            |-- entrypoint.py (generated by SDK if it does not exist)
+
+        In the built image, it will look like:
+        container_workdir/
+        |-- requirements.txt
+        |-- user_src_dir/
+            |-- predictor.py
+            |-- requirements.txt
+            |-- user_code/
+            |   |-- utils.py
+            |   |-- custom_package.tar.gz
+            |   |-- ...
+            |-- ...
+            |-- entrypoint.py
+
+        Note: If you want to install packages from *.tar.gz, you need to specify the
+        path relative to container workdir in requirements.txt. In this case you need
+        to in requirements.txt, specify `./user_src_dir/user_code/custom_package.tar.gz`.
+
+        Args:
+            src_dir (str):
+                Required. The path to the local directory including all needed files such as
+                predictor. The whole directory will be copied to the image.
+            output_image_uri (str):
+                Required. The image uri of the built image.
+            predictor (Type[Predictor]):
+                Optional. The custom predictor class consumed by handler to do prediction.
+            handler (Type[Handler]):
+                Required. The handler class to handle requests in the model server.
+            base_image (str):
```

Comment:
Can we also list the requirements of the base image here?

Like must have python & pip installed ... etc.

Reply:
Done.